javase/src/main/java/net/glxn/qrgen/javase/QRCode.java

```diff
@@ -1,167 +1,269 @@
-        return this;
-    }
-
-    /**
-     * Overrides the default charset by supplying a {@link com.google.zxing.EncodeHintType#CHARACTER_SET} hint to {@link
-     * com.google.zxing.qrcode.QRCodeWriter#encode}
-     *
-     * @param charset the charset as string, e.g. UTF-8
-     * @return the current QRCode object
-     */
-    public QRCode withCharset(String charset) {
-        return withHint(EncodeHintType.CHARACTER_SET, charset);
-    }
-
-    /**
-     * Overrides the default error correction by supplying a {@link com.google.zxing.EncodeHintType#ERROR_CORRECTION} hint to
-     * {@link com.google.zxing.qrcode.QRCodeWriter#encode}
-     *
-     * @param level the error correction level to use by {@link com.google.zxing.qrcode.QRCodeWriter#encode}
-     * @return the current QRCode object
-     */
-    public QRCode withErrorCorrection(ErrorCorrectionLevel level) {
-        return withHint(EncodeHintType.ERROR_CORRECTION, level);
-    }
-
-    /**
-     * Sets hint to {@link com.google.zxing.qrcode.QRCodeWriter#encode}
-     *
-     * @param hintType the hintType to set
-     * @param value the concrete value to set
-     * @return the current QRCode object
-     */
-    public QRCode withHint(EncodeHintType hintType, Object value) {
-        hints.put(hintType, value);
-        return this;
-    }
-
-    @Override
-    public File file() {
-        File file;
-        try {
-            file = createTempFile();
-            MatrixToImageWriter.writeToPath(createMatrix(text), imageType.toString(), file.toPath(), matrixToImageConfig);
-        } catch (Exception e) {
-            throw new QRGenerationException("Failed to create QR image from text due to underlying exception", e);
-        }
-
-        return file;
-    }
-
-    @Override
-    public File file(String name) {
-        File file;
-        try {
-            file = createTempFile(name);
-            MatrixToImageWriter.writeToPath(createMatrix(text), imageType.toString(), file.toPath(), matrixToImageConfig);
-        } catch (Exception e) {
-            throw new QRGenerationException("Failed to create QR image from text due to underlying exception", e);
-        }
-
-        return file;
-    }
-
-    @Override
-    protected void writeToStream(OutputStream stream) throws IOException, WriterException {
-        MatrixToImageWriter.writeToStream(createMatrix(text), imageType.toString(), stream, matrixToImageConfig);
-    }
-
-    private File createTempSvgFile() throws IOException {
-        return createTempSvgFile("QRCode");
-    }
-
-    private File createTempSvgFile(String name) throws IOException {
-        File file = File.createTempFile(name, ".svg");
-        file.deleteOnExit();
-        return file;
-    }
-
-    public QRCode withColor(int onColor, int offColor) {
-        matrixToImageConfig = new MatrixToImageConfig(onColor, offColor);
-        return this;
-    }
+	public static final MatrixToImageConfig DEFAULT_CONFIG = new MatrixToImageConfig();
```

Comment:
I would appreciate if the diff here only contained relevant changes. Seems your changes affect nearly every line in this file. Please try to keep the changes to lines that are actually changed, and not reformat the whole file.

Reply:
Ok